echopop/computation/biology.py

```diff
@@ -104,83 +128,200 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+    weight_strata_aged_unaged = pd.concat( [ haul_strata_weight , specimen_strata_weight ] )
+ 
     ### Sum weights for each stratum
     weight_strata = (
-            weight_strata_station
-            .groupby( [ 'stratum_num' ] )
-            .apply( lambda x: pd.Series( { 'weight_stratum_total': x.stratum_weight.sum( ) } ) )            
-            .reset_index( )
-        )
+        weight_strata_aged_unaged
+        .groupby( "stratum_num" )[ "stratum_weight" ]
+        .sum( )
+        .to_frame( 'weight_stratum_all' )
+        .reset_index( )
+    )
     
     ### Carriage return
-    return weight_strata_station , weight_strata
+    return weight_strata_aged_unaged , weight_strata
 
-def sum_strata_length_age_sex_weight( haul_data: pd.DataFrame ,
-                                      specimen_data: pd.DataFrame ,
-                                      length_intervals: np.ndarray ,
-                                      age_intervals: np.ndarray ):
-    
-    ### Sum weights from each station and strata 
-    weight_strata_station , weight_strata = sum_strata_weight( haul_data ,
-                                                               specimen_data )
-    
-    ### Calculate summed length-age-sex weight bins across strata
-    weight_length_age_sex_stratum = (
+def compute_index_aged_weight_proportions( specimen_data: pd.DataFrame ,
+                                           length_intervals: np.ndarray ,
+                                           age_intervals: np.ndarray ):
+    """
+    Calculate length-age binned weight proportions    
+
+    Parameters
+    ----------
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    length_intervals: np.ndarray
+        Array containing length bins/intervals
+    age_intervals: np.ndarray
+        Array containing age bins/intervals
+    """ 
+
+    ### Process the specimen data 
+    specimen_binned = (
         specimen_data
+        # ---- Drop unaged fish 
+        # !!! TODO: pending what FEAT says, weights associated with 
+        # missing ages should be added into the 'unaged' category. 
+        # This would further mean apportioning these into `weight_strata_aged_uanged`
         .dropna( how = 'any' )
-        .loc[ lambda x: x.sex != 'unsexed' ]
-        .pipe( lambda df: pd.concat( [ df , 
-                                       df.copy( ).assign( sex = 'all' ) ] ) )
+        # ---- Remove unsexed fish
+        .loc[ lambda df: df.sex != 'unsexed' ]
+        # ---- Bin length
+        .bin_variable( length_intervals , 'length' )
+        # ---- Age bin
         .bin_variable( age_intervals , 'age' )
-        .bin_variable( length_intervals , 'length' )      
-        .groupby( [ 'stratum_num' , 'age_bin' , 'length_bin' , 'sex' ] )
-        .apply( lambda df: pd.Series( { 'summed_weight_all': df.weight.sum( ) ,
-                                        'summed_weight_adult': df.loc[ df.age > 1 ].weight.sum( ) } ) ) 
+    )
```

Comment:
The lambda function could just be a row selection, and if I were to write it often I just write the variables out explicitly:
```python
specimen_binned = specimen_data[specimen_data["sex"]=="unsexed"].dropna(how="any", subset="...")
specimen_binned["length"] = specimen_binned.bin_variable(length_intervals, "length")
specimen_binned["age"] = specimen_binned.bin_variable(age_intervals, "age")
```

Reply:
And in what you have for `bin_variable`:
I would vectorize it to do:
```python
dataframe[f"{bin_variable}_bin"] = pd.cut(x[bin_variable], bin_values)
```
instead of using lambda functions.

Reply:
@leewujung I agree with vectorizing the `bin_variable` function (i.e. removing the `lambda` function component); however, there is an issue with implementing the first comment. One minor issue is that `bin_variable` is still associated with the monkey patch decorator: `@patch_method_to_DataFrame( pd.DataFrame )` ([#166](https://github.com/OSOceanAcoustics/echopop/issues/166)). So the expected output is, effectively, the updated dataframe with the newly assigned bin variables. If the monkey patch is removed, then this would require `pipe(...)`. The monkey patch aside, the current implement therefore precludes assigning the output of `specimen_bined.bin_variable( ... )` to a given column as exemplified in your comment. 

The original intent of the `bin_variable` function was to coincide with `bin_stats(...)` to be a bit more explicit in describing what was occurring at that point in the dataframe pipe. So maybe this is more of a style/preference thing? Functionally, `bin_variable` could be circumvented entirely here by: 

```python
specimen_binned = specimen_data[ specimen_data.sex !="unsexed" ].dropna( how = "any" , subset="age" ) 
specimen_binned[ "length_bin" ] = pd.cut( specimen_binned.length , length_intervals ) 
specimen_binned[ "age_bin" ] = pd.cut( specimen_binned.age , age_intervals ) 
```
So this may indicate that `bin_variable` could be deprecated (or removed completely). I actually think `bin_variable` could be refitted to bin over multiple variables to cut down on repetitiveness both here and elsewhere in the dataflow. I'll add a `TODO` to `bin_variable` to make a note that either way something should be changed or that the function can be deprecated and updates are required to other functions/operations that use `bin_variable`. Thoughts?

Reply:
Good question! It's often hard to determine if something ops should be factored out as a function or not. I think given the usecase of `bin_variable`, if I were to do it, I would likely go with the path of not adding a new function, because `pd.cut` is widely used, so the intention for the code is clear. Also because the code for using `pd.cut` directly short.